src/Illuminate/Foundation/Auth/VerifiesEmails.php

```diff
@@ -35,6 +35,9 @@ public function verify(Request $request)
         if ($request->route('id') != $request->user()->getKey()) {
             throw new AuthorizationException;
         }
+        if ($request->route('hash') != hash('sha1', $request->user()->email)) {
```

Comment:
Same issue. The email property might not exist, or be called that. We can only use methods from the contract, like getKey().